tdns/hsyncengine.go

```
@@ -330,104 +664,64 @@ func (ar *AgentRegistry) HandleStatusRequest(req SyncStatus) {
 	}
 
 	// Get current agents without waiting for any pending operations
-	agents := map[string]*Agent{}
+	agents := map[AgentId]*Agent{}
 	for _, agent := range ar.S.Items() {
 		// Make a clean copy of the agent for the response
-		sanitized := SanitizeForJSON(*agent) // Shallow copy
-		if foo, ok := sanitized.(*Agent); ok {
+		saneAgent := SanitizeForJSON(*agent) // Shallow copy
```

golangci-lint (1.64.8)

copylocks: call of SanitizeForJSON copies lock value: github.com/johanix/tdns/tdns.Agent contains sync.RWMutex

(govet)